usmqe/gluster/gluster.py

```diff
@@ -0,0 +1,208 @@
+        return output
+
+    def run_on_all_nodes(self, command, nodes=None, executor=None,
+                         parse_output=xml.etree.ElementTree.fromstring):
+        """
+        Run command on all gluster nodes
+        """
+        last_error = None
+        output = None
+        if nodes is None:
+            nodes = usmqe.inventory.role2hosts("gluster")
+        if not executor:
+            executor = self.cmd
+        for node in nodes:
+            try:
+                output = executor.run(node, command)
+            except GlusterCommandErrorException as err:
+                last_error = err
+                continue
+            break
+        else:
+            if last_error:
+                raise GlusterCommandErrorException(
+                    "Problem with gluster command '%s'.\n"
+                    "Last command rcode: %s, stdout: %s, stderr: %s" %
+                    (last_error.cmd, last_error.rcode,
+                     last_error.stdout, last_error.stderr))
+            else:
+                raise GlusterCommandErrorException(
+                    "Problem with gluster command '%s'.\n"
+                    "Possible problem is no gluster-node (gluster_node list: %s)" %
+                    (command, usmqe.inventory.role2hosts("gluster")))
+
+        if parse_output:
+            output = parse_output(output)
+        return output
+
+    def get_volume_names(self):
+        """
+        Returns name(s) of volume.
+        TODO specify order or some search if more volumes
+        """
+        vol_name = self.run_on_node(command="volume info").findtext(
+            "./volInfo/volumes/volume/name")
+        LOGGER.debug("Volume_name: %s" % vol_name)
+        return vol_name
+
+    def get_hosts_from_trusted_pool(self, host):
+        """
+        Returns host names from trusted pool with given hostname.
+        """
+        # TODO change to right path to hostnames
+        hosts = self.run_on_node(node=host, command="peer status").findall(
+            "./peerStatus/peer/hostnames/hostname")
+        hosts = [x.text for x in hosts]
+        hosts.append(host)
+        LOGGER.debug("Hosts in trusted pool: %s" % hosts)
+        return hosts
+
+    def find_volume_name(self, name, expected=True):
+        """
+        Check if there is volume with given name.
+        """
+        volume_name = self.get_volume_names()
+        found = False
+        # TODO test what vol_name looks like when there are more volumes name
+        if isinstance(volume_name, list):
+            for item in volume_name:
+                # TODO use pytest.check in if?
+                if item == name:
+                    found = True
+        else:
+            if volume_name == name:
+                found = True
+        if expected:
+            pytest.check(
+                found,
+                "{} should be among volumes from output \
+                of gluster volume info command".format(name))
```

Comment:
hmm, I forgot on expected here, so you could first put the one line expression to the found variable and then make the pytest check the same as before.